frontend/app/blueprints/groups/templates/shared.html

```diff
@@ -113,4 +113,169 @@
     const thresholdValue = thresholdElem.data
     document.getElementById("similar-samples-button").disabled = isNaN(parseInt(thresholdValue))
 });
-{% endmacro %}
\ No newline at end of file
+{% endmacro %}
+
+{% macro qc_bulk_toggle(bad_qc_actions) %}
+    {#
+
+        Button interacting with sample list in groups. Activates pop up dialog when used.
+
+        Inputs:
+            - bad_qc_actions : List[str]    List of selectable follow-up actions for failed samples
+
+    #}
+    <div id="qc-menu-dropdown" class="dropdown">
+        <button
+            id="toggle-qc-btn"
+            class="btn btn-sm btn-outline-secondary dropdown-toggle ms-4"
+            data-bs-toggle="dropdown" data-bs-auto-close="outside"
+            disabled
+        >
+            Set QC status
+        </button>
+        <form
+            class="needs-validation"
+            method="post"
+            id="qc-form-control"
+            action="{{ url_for('groups.update_qc_classification') }}"
+        >
+            <div id="qc-dropdown-contents" class="dropdown-menu dropdown-menu-end p-2">
+                <div class ="row pb-2">
+                    <div class="btn-group btn-group-sm col-auto" role="group" aria-label="Sample qc rejection group">
```

Comment:
Can the code for the passed/ failed QC, rejection reasons and comment be shared by the sample sidebar?

Reply:
Maybe. The sidebar macro requires sample-specific data, but maybe that can be worked around somehow.

I'll give it a try.

Reply:
Done! See new comments in main PR thread.